File: 算法/代码随想录算法PDF/00调试源代码/860.柠檬水找零.py
#
# @lc app=leetcode.cn id=860 lang=python3
#
# [860] 柠檬水找零
#

# @lc code=start
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        ten, five = 0, 0
        for bill in bills:
            if bill == 5:
                five += 1
            if bill == 10:
                if five:
                    five -= 1
                    ten += 1
                else:
                    return False
            if bill == 20:
                if five and ten:
                    five -= 1
                    ten -= 1
                elif five >= 3:
                    five -= 3
                else:
                    return False
        return True
# @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/860.柠檬水找零.cpp
/*
 * @lc app=leetcode.cn id=860 lang=cpp
 *
 * [860] 柠檬水找零
 */

// @lc code=start
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int ten = 0, five = 0;
        for (int i = 0; i < bills.size(); i++) {
            if (bills[i] == 5) five++;
            if (bills[i] == 10) {
                if (five) {
                    five--;
                    ten++;
                } else {
                    return false;
                }
            }
            if (bills[i] == 20) {
                if (ten && five) {
                    ten--;
                    five--;
                } else if (five >= 3) {
                    five -= 3;
                } else {
                    return false;
                }
            }
        }
        return true;
    }
};
// @lc code=end

